### PR TITLE
feat: add ReservedDataDiskSlotNum to helm-chart 

### LINKED ISSUE #2382: ReservedDataDiskSlotNum not available in helm chart
**Is your feature request related to a problem?/Why is this needed**
ReservedDataDiskSlotNum should be put into the helm-chart
ReservedDataDiskSlotNum is not given the the Driver struct from the DriverOptions struct

**Describe the solution you'd like in detail**
Being able to use it in the helmchart

### PR DESCRIPTION
**Pull request by ClementLachaussee**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

The option is not available in the HelmChart, and the option is never given from the DriverOptiont struct to Driver struct. 

**Which issue(s) this PR fixes**:

Fixes #2382

**Special notes for your reviewer**:

### CODE REVIEW

File: charts/latest/azuredisk-csi-driver/values.yaml [hunk 1]
```diff
@@ -168,6 +168,8 @@ driver:
   # maximum number of attachable volumes per node,
   # maximum number is defined according to node instance type by default(-1)
   volumeAttachLimit: -1
+  # reserved data disk slot number per node
+  reservedDataDiskSlotNum: 0
```
Comment (andyzhangx) [inline charts/latest/azuredisk-csi-driver/values.yaml:172]: could you move to `node.reservedDataDiskSlotNum`

**Comment by ClementLachaussee** [comment]
Hey @andyzhangx !
Thanks for your quick review, I have modified as requested and taken the liberty to modify the comment.

File: pkg/azuredisk/azuredisk.go [hunk 2]
```diff
@@ -145,6 +145,7 @@ func newDriverV1(options *DriverOptions) *Driver {
 	driver.Version = driverVersion
 	driver.NodeID = options.NodeID
 	driver.VolumeAttachLimit = options.VolumeAttachLimit
+	driver.ReservedDataDiskSlotNum = options.ReservedDataDiskSlotNum
```
Comment (andyzhangx) [inline pkg/azuredisk/azuredisk.go:148]: it seems this is a bug, `ReservedDataDiskSlotNum` never takes effect in the before?

Comment (andyzhangx) [reply pkg/azuredisk/azuredisk.go:148]: can you make this change as a separate PR? I will cherrypick to old releases, thx

Comment (ClementLachaussee) [reply pkg/azuredisk/azuredisk.go]: done in !2384

Comment (ClementLachaussee) [reply pkg/azuredisk/azuredisk.go]: No it doesn't after testing the CSIDriver object is never modified

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by andyzhangx** [comment]
> @andyzhangx do you know when this can be added to a release ?

@ClementLachaussee I am adding this chart config into v1.29.7 release since it does not have that ReservedDataDiskSlotNum assign bug:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/2395

**Pull request merged**